### ISSUE TITLE
[Boss Idea + integrated lore mechanics + other cool stuff <3]

### ISSUE THREAD

**Issue by PugLord** [AngelArenaAllstars/suggestions#96]
_Introduction_
In games like World of Warcraft, Diablo and Darksouls what makes a fight truely memorable? Is it the intense moment of clutch game-play and skill? Is it the lore behind a boss that makes you think "Wow that was a cool boss because [insert cool lore]"? Is it that feeling of clearing out a dungeon to get to a boss just to be beat multiple times until you try hard, get good and beat it with your friends? Yes all of these things make a boss even more enjoyable. 
The Medusa boss... everyone knows if you look at her you turn to stone. We can make this into a game mechanic by forcing the players to use methods of attacking her indirectly or unique tactics to avoid being turned to stone for instance one player leads her away whilst the others attack or the players can only see her by facing mirrors mentioned in your video. This boss has both a lore in which mechanics are derived and is easy to under stand for new players... don't look at it.

_Main Idea_   
So in conclusion a boss can have be made interesting using lore to flesh out game mechanics which is where I got the following cool ideas.
The Dune worm, giant worm the burrows under ground then pops up to attack players, easy to understand as it have been in quite a few games... so how could we flesh out mechanics from this? Well when the worm burrows it can display a red area at where it is gonna resurface. If the player doesn't move from this location the player will be consumed now whether or not the player dies instantly or is trapped in the beasts belly whilst taking damage is up to you. If the player moves out of the way the monster can use its main attack by spitting corrosive bile into the air and landing quickly (not too fast that the player can't dodge). The dune worm could also spit out enemies close by as an AOE and summon spell that could be canceled. The boss would take damage if attacked whilst surfaced.

This next boss idea should be a final boss and I will explain in a moment. Whilst playing a Warcraft 3  custom game I encountered something that I really liked. The custom game was about building up your empire to defeat your opponents with a few major npc cities near buy where you can sell goods etc. 
There was a shadow book a player could make to summon a calamity in which the player could choose which one they want to bring into their world. This calamity required the cooperation of multiple empires to work together to defeat it and if they couldn't it would be good game as the calamity would over power and defeat all the empires in the game. One of the calamities would summon a vampire queen at a castle on the map and every night she would got to the npc cities and 1 by 1 convert all the villagers into her vampire army. I am going to call this boss X as it could be anything. So after 45-60 min of game play each team will be allowed to enter a area (center of map or corner etc) or each team could work together to gather enough gold to build an item to open the area (50k-100k gold or something). 
The team that got into that area would have to fight boss x (Night-stalker a boss who if remains undefeated for too long will run rampant with an army of mini bosses [quality over quantity] or a Underlord who if remains undefeated will go to areas on the map to kill creeps to gain permanent bonus damage or a necro who if remains undefeated will slowly kill all players on the map). 
Boss x is basically a boss that if 1 team fails to beat both teams will have to work together against the clock to defeat him or not and both teams lose. The team that beats him single handed wins the game or gets a super powerful tool that gives the team a huge advantage. Now whats good about this is that if you say something like the dark lord has awakened or hell has been opened lore-wise there could be in game repercussions if one team loses the fight against it.

_Conclusion_   
You are doing a great job with AngelArenaAllstars and it might land up like dotA by becoming its own game which would be great. Keep it up and don't lose sight of your dreams.

**Comment by Haganeko** [comment]
Thank you for the post! It's nice to see a fellow Dune fan.
We're mostly looking for mechanics and design submission. If you could strip the flavor/lore from your ideas it would make them clearer and easier to read. We would be very thankful.
We have a final boss slot in the design. It's supposed to be a boss so hard it only gets beat in 0.1% or less of the games, and beating it wins the game outright. Baumi has put out more information on the way this boss would work. If you could flesh out your mechanics for this boss (we're pretty sure he'd be the Devil himself), we'd be happy to take them into consideration.

**Comment by PugLord** [comment]
In that case I have a great Idea for the demon lord (I will refer to the demon lord as doom).

When Doom is at 100% hp he will follow a attack pattern and since doom is right handed he will raise his sword into the air slowly then slash across the ground in an AOE attack, then doom will wait a second then doom will bring his sword over his left shoulder and slash the ground the other way. Doom will the finish his attack pattern by lifting the sword up high and slamming it down in front of him.
So hit box wise for his swings: 2 second animation delay, first swing horizontal hit box, 1 second animation delay, second swing horizontal hit box, 4 second delay (doom will try to start his vertical slash where most of the players are standing) vertical slash. Repeat attack cycle.
When doom is at 75% hp he will use blinding light forcing the players to the edge of the battle room to cast a Permanent Immolation on himself to burn near by players forcing the to cycle between tanks in the battle if they had any. Add in a visual effect where his Armour has started to burn so it becomes obvious what the spell has done. This spell takes 5 seconds to cast doom can be stunned in this time to increase the time spent casting to get more damage dealt. Repeat attack cycle.
When doom is at 50% hp he will use blinding light again and root the players against the edgy of the room keeping them away whilst he casts the spell. Once doom is done with the channeled spell he will stab himself in the chest with his sword bringing himself to 40% hp then pull his sword out which is one fire now. This means all of dooms attacks during his attack cycle now deal bonus fire damage and have extended range due to the flames this will make the hit boxes for the players to dodge bigger and his vertical slam attack will cast dragon slave each time.
When doom is at 25% hp he will use blinding light then root the players against the wall to open a portal to summon 2 demonic monsters to defend him (Warlock Golems) the Golems will have a swipe attack that the players can dodge and a auto attack the players can't dodge. (If the player is hit by the small hit box swipe by the golem the player will be silenced and then Doomed by doom... meaning they are punished and silenced)
When doom is at 10% hp he will slam his sword into the ground doing a small AOE around him and then pull his sword out of the ground (the sword loses its fire effect and AOE boost and dragon slave extra on vertical slam attack) but this will set the ground in the arena on fire and give doom a shield. This is the critical moment in the fight where the attacking team must decide whether to finish the fight or retreat.
If they win the winning team can get an enchantment on all weapons to deal bonus fire damage and the demon lord will drop his sword which any melee hero can pick up to deal insane damage (acts like divine rapier and drops on death).
If the players are defeated or retreat doom will leave his arena to start consuming the trees on the map a message will be sent to both teams saying "The Demon Lord Has Entered Our World". 
A counter will appear for both teams showing how many trees have been consumed by the demon lord.
Each time the demon lord consumes a tree he will gain back lost hp, his consume tree ability has a 2 second cool down. Doom will still have his burning armor, crimson guard shield and a circle filled with fire around him. Doom will summon a demonic monster to aid him after every 10 trees he eats (so 1 golem every 20 seconds). The team that kills him will get the buff and sword. The trees can be changed to any kind of objective for the doom lord to destroy or consume.

That's it. Hope you like it.

**Comment by Haganeko** [comment]
Thank you for the idea, and sorry for the wait.
This looks interesting, but it could use some formatting and cleaning up. #37 has a useful template for this. Focusing more on mechanics instead of fluff could also help with clarity.
In terms of mechanics, the doomsday idea sounds like a cheap way for a losing team to cause havoc. I don't like it much.
If and when you review this, please open a new issue with the reviewed version.
